### ISSUE TITLE
This website URL opens but KR gives a 403 error

### ISSUE THREAD

**Issue by DaveF63** [keepright/keepright#78]
Hi
https://www.keepright.at/report_map.php?zoom=18&lat=51.38798&lon=-2.36332&layers=B0T&ch=0%2C411&show_ign=0&show_tmpign=0

This is just one example of many. What is KR objecting to?

**Comment by mueschel** [comment]
Hi,
I just checked the website with one of those "is my website reachable' tools. It tested reachability from various places and about 60% of them also reported a 403 error. I guess this is a real issue with this website.

**Comment by DaveF63** [comment]
Hmm... OK:
1.  What were the "is my website reachable" tools
2. Note I said: "This is just one example of many"
3. It loads by clicking on the URL within KR's popup
4. It loads with IE & firefox, two *very* popular browsers.

So my question of why KR flags a problem still stands.

**Comment by mueschel** [comment]
It loads from your PC and from mine, but many other can not load the page. E.g. got to https://www.uptrends.com and enter the URL there. It checks if the web page is actually reachable from many locations world-wide. This is neither a problem of browsers nor with keepright, this is a problem with the web server itself.

**Comment by DaveF63** [comment]
The URL in OSM is written correctly.
Are KR users able to do anything to effect how these erroneous web servers deal with this URL?
If not, why is KR flagging it as if they can?

**Comment by mueschel** [comment]
In this special case you can't do anything about it, despite informing the web master of this web page. But I guess they won't care about problems accessing the page from other countries...  You can just flag this issue as false positive.
In many other cases, you can do something about it, so the error message can be useful. I checked some of the spots nearby:

https://www.openstreetmap.org/browse/way/478357704  - I get a 403 as well, the link seems to be wrong
https://www.openstreetmap.org/browse/way/120543184 - Keepright is correct, the server gives back an error (but displays the page nevertheless)
https://www.openstreetmap.org/browse/way/216978938 - definitely wrong
https://www.openstreetmap.org/browse/way/431372811 - looks fine. check if keepright insists on it
https://www.openstreetmap.org/browse/node/1685718558 - website not available
https://www.openstreetmap.org/browse/way/55380657 - I can't check it. I'm redirected to a German web page and Keepright is likely redirected to an Austrian one... no way to check when you're not in the UK (damn webmasters).

**Comment by DaveF63** [comment]
Thanks for looking into this. I've gone though all my local ones & it was about 50% erroneous URLs which I was able to fix & 50% gave me no errors. Would it be beneficial to split error checking down to individual error codes? 403,404, 503 etc?

Is there maybe a way to localize the checking area? The /vast/ majority of URL clicks for UK establishments would be from within the UK & likewise with other countries.    

> You can just flag this issue as false positive.
False positive popups reappear after a few weeks. Is there a reason for this? I assume these are stored locally. Is there a way for me to ensure they're permanently hidden?

Has #63 been implemented?

**Comment by mueschel** [comment]
> Would it be beneficial to split error checking down to individual error codes? 403,404, 503 etc?

I don't think this helps. The action by the user is always the same: Check if the URL is correct or if there is a better link to put in the key.

> Is there maybe a way to localize the checking area? The /vast/ majority of URL clicks for UK establishments would be from within the UK & likewise with other countries.

No, this can't be done without having a server or proxy located in the respective area.

> Has #63 been implemented?

No, the issue is still open.
